kombu/utils/url.py

```diff
@@ -97,22 +96,19 @@ def as_url(scheme, host=None, port=None, user=None, password=None,
     return ''.join(str(part) for part in parts if part)
 
 
-def sanitize_url(url, mask='**'):
-    # type: (str, str) -> str
+def sanitize_url(url: str, mask: str='**') -> str:
     """Return copy of URL with password removed."""
     return as_url(*_parse_url(url), sanitize=True, mask=mask)
 
-
-def maybe_sanitize_url(url, mask='**'):
-    # type: (Any, str) -> Any
+_T = TypeVar("_T")
+def maybe_sanitize_url(url: _T, mask: str='**') -> Union[_T, str]:
```

Comment:
```suggestion
_T = TypeVar("_T")

@overload
def maybe_sanitize_url(url: str, mask: str = ...) -> str:
    ...
    
@overload
def maybe_sanitize_url(url: _T, mask: str = ...) -> _T:
    ...

def maybe_sanitize_url(url, mask='**'):
```

Reply:
Actually, the `str -> str` case is already covered by `_T -> _T`, so it it is not required. Should I specify it anyway ?

Reply:
Oh! Then no. 😅